### PR TITLE
Add Taxprofiler ordering support

### PR DESCRIPTION
**Pull request by islean**
## Description

Updates the 1508 order form to version 32 wherein Taxprofiler is orderable

### Added

- Taxprofiler excel order form test fixture

### Changed

- 1508 orderforms need to be version 32

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/apps/orderform/excel_orderform_parser.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ class ExcelOrderformParser(OrderformParser):
     model_config = ConfigDict(arbitrary_types_allowed=True)
     NO_ANALYSIS: str = "no-analysis"
     NO_VALUE: str = "no_value"
-    SHEET_NAMES: list[str] = ["Orderform", "orderform", "order form"]
+    SHEET_NAMES: list[str] = ["Orderform", "orderform", "order form", "Order Form"]
```
Comment (islean) [inline cg/apps/orderform/excel_orderform_parser.py:25]: The sheet containing all the sample data has changed name from "order form" to "Order Form". This code broke without this modification. Definitely a code smell that this is needed but probably not something to be fixed here?

**Comment by islean** [comment]
Order submitted using excel sheet:

image

**Comment by islean** [comment]
Case created:

image

Tied to the samples:

image

Samples have data:

image

Order created and tied to the case:

image

**Comment by islean** [comment]
Added to LIMS: 
image

**Comment by islean** [comment]
Sample data added in lims 
image

File: cg/apps/orderform/excel_orderform_parser.py (hunk 1 above)
Comment (ChrOertlin) [reply cg/apps/orderform/excel_orderform_parser.py:25]: Ideally the orderforms get streamlined and conform to a specific sheet name. I would suggest the order form gets fixed rather us adapting to it here

**Review by ChrOertlin — APPROVED** [review]
Looks good!

File: cg/apps/orderform/excel_orderform_parser.py (hunk 1 above)
Comment (islean) [reply cg/apps/orderform/excel_orderform_parser.py:25]: While I do agree, there is a deadline to meet this afternoon so I would rather this be done when there is less time pressure

File: cg/constants/orderforms.py [hunk 2]
```diff
@@ -30,12 +30,13 @@ class Orderform(StrEnum):
...
     MICROBIAL_FASTQ: str = "microbial_sequencing"
     PACBIO_LONG_READ: str = "pacbio_revio_sequencing"
+    TAXPROFILER: str = "1508"
 
     @staticmethod
     def get_current_orderform_version(order_form: str) -> str:
         """Returns the current version of the given order form."""
         current_order_form_versions = {
-            Orderform.MIP_DNA: "31",
+            Orderform.MIP_DNA: "32",
```
[the remark this answers is not in the document]
Comment (islean) [reply cg/constants/orderforms.py:39]: Orderform.MIP_DNA == Orderform.Taxprofiler == 1508. I can add to the docstring, but in some sense the parser would need some more extensive reworks to be logical.

[the remark this answers is not in the document]
Comment (islean) [reply cg/constants/orderforms.py:39]: Added clearer docstring

**Pull request merged**